src/libsync/discoveryphase.h

```diff
@@ -138,20 +106,22 @@ class DiscoverySingleDirectoryJob : public QObject
     void setIsRootPath() { _isRootPath = true; }
     void start();
     void abort();
+    std::vector<std::unique_ptr<csync_file_stat_t>> &&takeResults() { return std::move(_results); }
+
```

Comment:
Although the API is more familiar, the issue with Qt containers is that they don't support rvalue-ref emplacing, which means I'd have to use a `QSharedPointer` instead of a `unique_ptr` (since `unique_ptr` isn't copyable). And we don't need the shared pointer, I like the very clear ownership transfers of `unique_ptr` through `std::move`.